state/core_access.go

```diff
@@ -174,24 +192,74 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		appblobs[i] = &b.Blob
 	}
 
-	response, err := appblob.SubmitPayForBlob(
-		ctx,
-		ca.signer,
-		ca.coreConn,
-		appblobs,
-		apptypes.SetGasLimit(gasLim),
-		withFee(fee),
-	)
-	// metrics should only be counted on a successful PFD tx
-	if err == nil && response.Code == 0 {
-		ca.lastPayForBlob = time.Now().UnixMilli()
-		ca.payForBlobCount++
+	// if the min gas price is not set we query the node that we are connected for it
+	if ca.minGasPrice < 0 {
+		minGasPrice, err := ca.queryMinimumGasPrice(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("querying minimum gas price: %w", err)
+		}
+		ca.minGasPrice = minGasPrice
+	}
```

Comment:
This should be synchronized with singleflight protection as SubmitPayForBlob can be called from multiple routines.
(does not the case because no nounce handling yet, but we want concurrent PFBs in the future)